### PR TITLE
Update build config and dependencies

### PR DESCRIPTION
**Pull request by kenlheritier**
- Update AGP
- Update plugins and dependencies

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -28,12 +28,12 @@ android {
 }
 
 dependencies {
-    implementation "io.didomi.sdk:android:1.86.3"
+    implementation "io.didomi.sdk:android:1.87.0"
 
     implementation "androidx.appcompat:appcompat:1.6.1"
     //noinspection GradleDependency - Higher version will require kotlin
```
Comment (nicolas-chaix-didomi) [inline app/build.gradle]: you can remove this comment, gradle will suggest you a new version

File: gradle.properties [hunk 2]
```diff
@@ -19,3 +19,5 @@ android.useAndroidX=true
 # resources declared in the library itself and none from the library's dependencies,
 # thereby reducing the size of the R class for that library
 android.nonTransitiveRClass=true
+android.defaults.buildfeatures.buildconfig=true
```
Comment (nicolas-chaix-didomi) [inline gradle.properties]: You can remove this line: its usage it deprecated and you don't use build config fields if I'm correct

File: gradle/wrapper/gradle-wrapper.properties [hunk 3]
```diff
@@ -1,6 +1,6 @@
 #Fri Nov 25 11:23:49 CET 2022
 distributionBase=GRADLE_USER_HOME
-distributionUrl=https\://services.gradle.org/distributions/gradle-7.5-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-8.2-bin.zip
```
Comment (nicolas-chaix-didomi) [inline gradle/wrapper/gradle-wrapper.properties:3]: Did you use the assistant for the migration? i'm surprised to see only the properties file changed

Comment (kenlheritier) [reply gradle/wrapper/gradle-wrapper.properties:3]: Yes I did, even took a screenshot before doing so:

Capture d’écran 2023-12-19 à 13 54 55

File: app/build.gradle (hunk 1 above)
Comment (kenlheritier) [reply app/build.gradle]: True, I updated the dependency: https://github.com/didomi/sample-app-android-java/pull/3/commits/c8d689f710f1125a22698cc3378942d008c7ad72

File: gradle.properties (hunk 2 above)
Comment (kenlheritier) [reply gradle.properties]: Done: https://github.com/didomi/sample-app-android-java/pull/3/commits/08332823e7109d915e0637650af408327cc307f2

**Review by nicolas-chaix-didomi — APPROVED** [review]

**Pull request merged**